### ISSUE TITLE
AzPainter crashed on initializing "Transformation" window

### ISSUE THREAD

**Issue by Symbian9** [Symbian9/azpainter#66]
### System info

- **Version:** `2.1.4`
- **OS/Platform:** Debian 9.x (x86_64)

### Reproducing 

1. Launch *AzPainter* with custom locale:
    ```
    $ azpainter --trfile /usr/share/azpainter/tr/ru.mtr
    ```
2. Select «Rectangle editing» tool;
3. Select rectangle region;
4. Click «Transformation» option icon on «Rectangle editing» toolbar to open «Transformation» window. 

### Actual behavior

*AzPainter* immediatly crashed on initializing «Transformation» window:

<details>
    <summary>Terminal Log</summary>

```
$ azpainter --trfile /usr/share/azpainter/tr/ru.mtr
*** Error in `azpainter': corrupted size vs. prev_size: 0x00005644652e4fd0 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x70bfb)[0x7ff299e5dbfb]
/lib/x86_64-linux-gnu/libc.so.6(+0x76fc6)[0x7ff299e63fc6]
/lib/x86_64-linux-gnu/libc.so.6(+0x773b8)[0x7ff299e643b8]
/lib/x86_64-linux-gnu/libc.so.6(+0x77ed8)[0x7ff299e64ed8]
/usr/lib/x86_64-linux-gnu/libX11.so.6(XFreeGC+0x93)[0x7ff29b8af223]
/usr/lib/x86_64-linux-gnu/libX11.so.6(XCreateBitmapFromData+0xd6)[0x7ff29b8a9b76]
azpainter(+0xa7c77)[0x56446316ac77]
azpainter(+0x7bae)[0x5644630cabae]
azpainter(+0x73e22)[0x564463136e22]
azpainter(+0x7330b)[0x56446313630b]
azpainter(+0xa8a16)[0x56446316ba16]
azpainter(+0x7886e)[0x56446313b86e]
azpainter(+0x987df)[0x56446315b7df]
azpainter(+0x72387)[0x564463135387]
azpainter(+0xab55)[0x5644630cdb55]
azpainter(+0x60ce5)[0x564463123ce5]
azpainter(+0xa8a16)[0x56446316ba16]
azpainter(+0x78806)[0x56446313b806]
azpainter(+0x7953)[0x5644630ca953]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1)[0x7ff299e0d2e1]
azpainter(+0x7a7a)[0x5644630caa7a]
======= Memory map: ========
5644630c3000-5644631b0000 r-xp 00000000 08:01 818335                     /usr/bin/azpainter
5644633af000-5644633b1000 r--p 000ec000 08:01 818335                     /usr/bin/azpainter
5644633b1000-5644633b2000 rw-p 000ee000 08:01 818335                     /usr/bin/azpainter
5644633b2000-5644633b3000 rw-p 00000000 00:00 0 
5644651b3000-5644652f7000 rw-p 00000000 00:00 0                          [heap]
7ff290000000-7ff290021000 rw-p 00000000 00:00 0 
7ff290021000-7ff294000000 ---p 00000000 00:00 0 
7ff29595f000-7ff295975000 r-xp 00000000 08:01 526694                     /lib/x86_64-linux-gnu/libgcc_s.so.1
7ff295975000-7ff295b74000 ---p 00016000 08:01 526694                     /lib/x86_64-linux-gnu/libgcc_s.so.1
7ff295b74000-7ff295b75000 r--p 00015000 08:01 526694                     /lib/x86_64-linux-gnu/libgcc_s.so.1
7ff295b75000-7ff295b76000 rw-p 00016000 08:01 526694                     /lib/x86_64-linux-gnu/libgcc_s.so.1
7ff295b95000-7ff295f7d000 rw-s 00000000 00:05 10813498                   /SYSV00000000 (deleted)
7ff295f7d000-7ff296365000 rw-s 00000000 00:05 10747960                   /SYSV00000000 (deleted)
7ff296365000-7ff296450000 rw-p 00000000 00:00 0 
7ff296450000-7ff29646e000 rw-s 00000000 00:05 10715191                   /SYSV00000000 (deleted)
7ff29646e000-7ff2964da000 r--p 00000000 08:01 531476                     /home/mint/.fonts/nohindisnr60.ttf
7ff2964da000-7ff296546000 r--p 00000000 08:01 531476                     /home/mint/.fonts/nohindisnr60.ttf
7ff296546000-7ff29654b000 r-xp 00000000 08:01 1321055                    /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
7ff29654b000-7ff29674a000 ---p 00005000 08:01 1321055                    /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
7ff29674a000-7ff29674b000 r--p 00004000 08:01 1321055                    /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
7ff29674b000-7ff29674c000 rw-p 00005000 08:01 1321055                    /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
7ff29674c000-7ff296755000 r-xp 00000000 08:01 1321075                    /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7ff296755000-7ff296954000 ---p 00009000 08:01 1321075                    /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7ff296954000-7ff296955000 r--p 00008000 08:01 1321075                    /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7ff296955000-7ff296956000 rw-p 00009000 08:01 1321075                    /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7ff296956000-7ff296960000 r-xp 00000000 08:01 1321047                    /usr/lib/x86_64-linux-gnu/libXcursor.so.1.0.2
7ff296960000-7ff296b5f000 ---p 0000a000 08:01 1321047                    /usr/lib/x86_64-linux-gnu/libXcursor.so.1.0.2
7ff296b5f000-7ff296b60000 r--p 00009000 08:01 1321047                    /usr/lib/x86_64-linux-gnu/libXcursor.so.1.0.2
7ff296b60000-7ff296b61000 rw-p 0000a000 08:01 1321047                    /usr/lib/x86_64-linux-gnu/libXcursor.so.1.0.2
7ff296b61000-7ff298f6f000 r--p 00000000 08:01 1048999                    /usr/lib/locale/locale-archive
7ff298f6f000-7ff298f82000 r-xp 00000000 08:01 526657                     /lib/x86_64-linux-gnu/libbsd.so.0.8.3
7ff298f82000-7ff299182000 ---p 00013000 08:01 526657                     /lib/x86_64-linux-gnu/libbsd.so.0.8.3
7ff299182000-7ff299183000 r--p 00013000 08:01 526657                     /lib/x86_64-linux-gnu/libbsd.so.0.8.3
7ff299183000-7ff299184000 rw-p 00014000 08:01 526657                     /lib/x86_64-linux-gnu/libbsd.so.0.8.3
7ff299184000-7ff299185000 rw-p 00000000 00:00 0 
7ff299185000-7ff29918a000 r-xp 00000000 08:01 1321051                    /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7ff29918a000-7ff299389000 ---p 00005000 08:01 1321051                    /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7ff299389000-7ff29938a000 r--p 00004000 08:01 1321051                    /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7ff29938a000-7ff29938b000 rw-p 00005000 08:01 1321051                    /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7ff29938b000-7ff29938e000 r-xp 00000000 08:01 1321040                    /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7ff29938e000-7ff29958d000 ---p 00003000 08:01 1321040                    /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7ff29958d000-7ff29958e000 r--p 00002000 08:01 1321040                    /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7ff29958e000-7ff29958f000 rw-p 00003000 08:01 1321040                    /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7ff29958f000-7ff299595000 r-xp 00000000 08:01 531647                     /lib/x86_64-linux-gnu/libuuid.so.1.3.0
7ff299595000-7ff299794000 ---p 00006000 08:01 531647                     /lib/x86_64-linux-gnu/libuuid.so.1.3.0
7ff299794000-7ff299795000 r--p 00005000 08:01 531647                     /lib/x86_64-linux-gnu/libuuid.so.1.3.0
7ff299795000-7ff299796000 rw-p 00006000 08:01 531647                     /lib/x86_64-linux-gnu/libuuid.so.1.3.0
7ff299796000-7ff2997bd000 r-xp 00000000 08:01 526682                     /lib/x86_64-linux-gnu/libexpat.so.1.6.2
7ff2997bd000-7ff2999bd000 ---p 00027000 08:01 526682                     /lib/x86_64-linux-gnu/libexpat.so.1.6.2
7ff2999bd000-7ff2999bf000 r--p 00027000 08:01 526682                     /lib/x86_64-linux-gnu/libexpat.so.1.6.2
7ff2999bf000-7ff2999c0000 rw-p 00029000 08:01 526682                     /lib/x86_64-linux-gnu/libexpat.so.1.6.2
7ff2999c0000-7ff2999c3000 r-xp 00000000 08:01 571261                     /lib/x86_64-linux-gnu/libdl-2.24.so
7ff2999c3000-7ff299bc2000 ---p 00003000 08:01 571261                     /lib/x86_64-linux-gnu/libdl-2.24.so
7ff299bc2000-7ff299bc3000 r--p 00002000 08:01 571261                     /lib/x86_64-linux-gnu/libdl-2.24.so
7ff299bc3000-7ff299bc4000 rw-p 00003000 08:01 571261                     /lib/x86_64-linux-gnu/libdl-2.24.so
7ff299bc4000-7ff299bec000 r-xp 00000000 08:01 1311697                    /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7ff299bec000-7ff299deb000 ---p 00028000 08:01 1311697                    /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7ff299deb000-7ff299dec000 r--p 00027000 08:01 1311697                    /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7ff299dec000-7ff299ded000 rw-p 00028000 08:01 1311697                    /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7ff299ded000-7ff299f82000 r-xp 00000000 08:01 571255                     /lib/x86_64-linux-gnu/libc-2.24.so
7ff299f82000-7ff29a182000 ---p 00195000 08:01 571255                     /lib/x86_64-linux-gnu/libc-2.24.so
7ff29a182000-7ff29a186000 r--p 00195000 08:01 571255                     /lib/x86_64-linux-gnu/libc-2.24.so
7ff29a186000-7ff29a188000 rw-p 00199000 08:01 571255                     /lib/x86_64-linux-gnu/libc-2.24.so
7ff29a188000-7ff29a18c000 rw-p 00000000 00:00 0 
7ff29a18c000-7ff29a1a4000 r-xp 00000000 08:01 571274                     /lib/x86_64-linux-gnu/libpthread-2.24.so
7ff29a1a4000-7ff29a3a3000 ---p 00018000 08:01 571274                     /lib/x86_64-linux-gnu/libpthread-2.24.so
7ff29a3a3000-7ff29a3a4000 r--p 00017000 08:01 571274                     /lib/x86_64-linux-gnu/libpthread-2.24.so
7ff29a3a4000-7ff29a3a5000 rw-p 00018000 08:01 571274                     /lib/x86_64-linux-gnu/libpthread-2.24.so
7ff29a3a5000-7ff29a3a9000 rw-p 00000000 00:00 0 
7ff29a3a9000-7ff29a3b8000 r-xp 00000000 08:01 1321063                    /usr/lib/x86_64-linux-gnu/libXi.so.6.1.0
7ff29a3b8000-7ff29a5b7000 ---p 0000f000 08:01 1321063                    /usr/lib/x86_64-linux-gnu/libXi.so.6.1.0
7ff29a5b7000-7ff29a5b8000 r--p 0000e000 08:01 1321063                    /usr/lib/x86_64-linux-gnu/libXi.so.6.1.0
7ff29a5b8000-7ff29a5b9000 rw-p 0000f000 08:01 1321063                    /usr/lib/x86_64-linux-gnu/libXi.so.6.1.0
7ff29a5b9000-7ff29a6bc000 r-xp 00000000 08:01 571263                     /lib/x86_64-linux-gnu/libm-2.24.so
7ff29a6bc000-7ff29a8bb000 ---p 00103000 08:01 571263                     /lib/x86_64-linux-gnu/libm-2.24.so
7ff29a8bb000-7ff29a8bc000 r--p 00102000 08:01 571263                     /lib/x86_64-linux-gnu/libm-2.24.so
7ff29a8bc000-7ff29a8bd000 rw-p 00103000 08:01 571263                     /lib/x86_64-linux-gnu/libm-2.24.so
7ff29a8bd000-7ff29a8c4000 r-xp 00000000 08:01 571276                     /lib/x86_64-linux-gnu/librt-2.24.so
7ff29a8c4000-7ff29aac3000 ---p 00007000 08:01 571276                     /lib/x86_64-linux-gnu/librt-2.24.so
7ff29aac3000-7ff29aac4000 r--p 00006000 08:01 571276                     /lib/x86_64-linux-gnu/librt-2.24.so
7ff29aac4000-7ff29aac5000 rw-p 00007000 08:01 571276                     /lib/x86_64-linux-gnu/librt-2.24.so
7ff29aac5000-7ff29aade000 r-xp 00000000 08:01 526821                     /lib/x86_64-linux-gnu/libz.so.1.2.8
7ff29aade000-7ff29acdd000 ---p 00019000 08:01 526821                     /lib/x86_64-linux-gnu/libz.so.1.2.8
7ff29acdd000-7ff29acde000 r--p 00018000 08:01 526821                     /lib/x86_64-linux-gnu/libz.so.1.2.8
7ff29acde000-7ff29acdf000 rw-p 00019000 08:01 526821                     /lib/x86_64-linux-gnu/libz.so.1.2.8
7ff29acdf000-7ff29ad48000 r-xp 00000000 08:01 1321816                    /usr/lib/x86_64-linux-gnu/libjpeg.so.62.2.0
7ff29ad48000-7ff29af48000 ---p 00069000 08:01 1321816                    /usr/lib/x86_64-linux-gnu/libjpeg.so.62.2.0
7ff29af48000-7ff29af49000 r--p 00069000 08:01 1321816                    /usr/lib/x86_64-linux-gnu/libjpeg.so.62.2.0
7ff29af49000-7ff29af4a000 rw-p 0006a000 08:01 1321816                    /usr/lib/x86_64-linux-gnu/libjpeg.so.62.2.0
7ff29af4a000-7ff29af7b000 r-xp 00000000 08:01 1316957                    /usr/lib/x86_64-linux-gnu/libpng16.so.16.28.0
7ff29af7b000-7ff29b17b000 ---p 00031000 08:01 1316957                    /usr/lib/x86_64-linux-gnu/libpng16.so.16.28.0
7ff29b17b000-7ff29b17c000 r--p 00031000 08:01 1316957                    /usr/lib/x86_64-linux-gnu/libpng16.so.16.28.0
7ff29b17c000-7ff29b17d000 rw-p 00032000 08:01 1316957                    /usr/lib/x86_64-linux-gnu/libpng16.so.16.28.0
7ff29b17d000-7ff29b22c000 r-xp 00000000 08:01 1311681                    /usr/lib/x86_64-linux-gnu/libfreetype.so.6.15.0
7ff29b22c000-7ff29b42b000 ---p 000af000 08:01 1311681                    /usr/lib/x86_64-linux-gnu/libfreetype.so.6.15.0
7ff29b42b000-7ff29b432000 r--p 000ae000 08:01 1311681                    /usr/lib/x86_64-linux-gnu/libfreetype.so.6.15.0
7ff29b432000-7ff29b433000 rw-p 000b5000 08:01 1311681                    /usr/lib/x86_64-linux-gnu/libfreetype.so.6.15.0
7ff29b433000-7ff29b475000 r-xp 00000000 08:01 1316964                    /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.12.0
7ff29b475000-7ff29b674000 ---p 00042000 08:01 1316964                    /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.12.0
7ff29b674000-7ff29b676000 r--p 00041000 08:01 1316964                    /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.12.0
7ff29b676000-7ff29b677000 rw-p 00043000 08:01 1316964                    /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.12.0
7ff29b677000-7ff29b688000 r-xp 00000000 08:01 1321053                    /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7ff29b688000-7ff29b887000 ---p 00011000 08:01 1321053                    /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7ff29b887000-7ff29b888000 r--p 00010000 08:01 1321053                    /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7ff29b888000-7ff29b889000 rw-p 00011000 08:01 1321053                    /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7ff29b889000-7ff29b9c3000 r-xp 00000000 08:01 1321034                    /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7ff29b9c3000-7ff29bbc3000 ---p 0013a000 08:01 1321034                    /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7ff29bbc3000-7ff29bbc4000 r--p 0013a000 08:01 1321034                    /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7ff29bbc4000-7ff29bbc9000 rw-p 0013b000 08:01 1321034                    /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7ff29bbc9000-7ff29bbec000 r-xp 00000000 08:01 571200                     /lib/x86_64-linux-gnu/ld-2.24.so
7ff29bbec000-7ff29bbf6000 rw-s 00000000 00:05 10682422                   /SYSV00000000 (deleted)
7ff29bbf6000-7ff29bc62000 r--p 00000000 08:01 531476                     /home/mint/.fonts/nohindisnr60.ttf
7ff29bc62000-7ff29bc69000 r--s 00000000 08:01 1319226                    /usr/lib/x86_64-linux-gnu/gconv/gconv-modules.cache
7ff29bc69000-7ff29bc75000 r--s 00000000 08:01 269277                     /var/cache/fontconfig/7ce6fd4d-b9ca-484d-a6ca-a524c4c55250-le64.cache-7
7ff29bc75000-7ff29bce9000 rw-p 00000000 00:00 0 
7ff29bcea000-7ff29bceb000 rw-p 00000000 00:00 0 
7ff29bceb000-7ff29bced000 r--s 00000000 08:01 269261                     /var/cache/fontconfig/735530d0-e3c8-46c2-b5ae-dcb0b7e57adf-le64.cache-7
7ff29bced000-7ff29bcf0000 r--s 00000000 08:01 269257                     /var/cache/fontconfig/c7917b08-db7c-4b11-b4c1-d36df51685de-le64.cache-7
7ff29bcf0000-7ff29bcf8000 r--s 00000000 08:01 269119                     /var/cache/fontconfig/76f45384-8719-4624-ade2-953ffe4adb13-le64.cache-7
7ff29bcf8000-7ff29bcfc000 r--s 00000000 08:01 269041                     /var/cache/fontconfig/e2d65a44-718d-47dc-bbeb-408ea5713221-le64.cache-7
7ff29bcfc000-7ff29bd25000 r--s 00000000 08:01 272457                     /var/cache/fontconfig/1e0b1a2a-c76e-470f-b84f-e6fc613a4575-le64.cache-7
7ff29bd25000-7ff29bd60000 r--s 00000000 08:01 268808                     /var/cache/fontconfig/d80c4535-d0a4-4f63-af16-1425885d41b7-le64.cache-7
7ff29bd60000-7ff29bd64000 r--s 00000000 08:01 268807                     /var/cache/fontconfig/d7fb5404-ed9a-46d2-99e5-90b060c21362-le64.cache-7
7ff29bd64000-7ff29bd77000 r--s 00000000 08:01 268334                     /var/cache/fontconfig/e27c236e-b4db-47c0-8dcb-4e263f295fd2-le64.cache-7
7ff29bd77000-7ff29bdc3000 r--s 00000000 08:01 569017                     /home/mint/.cache/fontconfig/3e7379bb-36b9-41aa-8a4c-68b8b08ef117-le64.cache-7
7ff29bdc3000-7ff29bdcd000 rw-p 00000000 00:00 0 
7ff29bdcd000-7ff29bdce000 r--s 00000000 08:01 266628                     /var/cache/fontconfig/b3db23e6-d3bb-44ab-abc4-2b37995b15e4-le64.cache-7
7ff29bdce000-7ff29bdd5000 r--s 00000000 08:01 268335                     /var/cache/fontconfig/481c2224-4918-4542-a31b-f6293440a43b-le64.cache-7
7ff29bdd5000-7ff29bdd6000 r--s 00000000 08:01 268132                     /var/cache/fontconfig/e66cce7b-f253-4edb-b958-5452927aa27a-le64.cache-7
7ff29bdd6000-7ff29bde2000 r--s 00000000 08:01 268125                     /var/cache/fontconfig/818131b0-465d-4ee8-87eb-311a304b79ca-le64.cache-7
7ff29bde2000-7ff29bde6000 r--s 00000000 08:01 268124                     /var/cache/fontconfig/2d96af55-af00-44b9-ae26-9a6e28b6d9fe-le64.cache-7
7ff29bde6000-7ff29bde7000 r--s 00000000 08:01 268119                     /var/cache/fontconfig/a42dc3a4-67aa-42a4-8669-70325fe60c55-le64.cache-7
7ff29bde7000-7ff29bdea000 r--s 00000000 08:01 266540                     /var/cache/fontconfig/d57e3dd8-27b1-4096-8608-63bc1455da38-le64.cache-7
7ff29bdea000-7ff29bdec000 r--s 00000000 08:01 268084                     /var/cache/fontconfig/fb540c3c-7fdd-4855-b355-ed1ba01b8bdc-le64.cache-7
7ff29bdec000-7ff29bded000 r--p 00023000 08:01 571200                     /lib/x86_64-linux-gnu/ld-2.24.so
7ff29bded000-7ff29bdee000 rw-p 00024000 08:01 571200                     /lib/x86_64-linux-gnu/ld-2.24.so
7ff29bdee000-7ff29bdef000 rw-p 00000000 00:00 0 
7ffdf1457000-7ffdf1478000 rw-p 00000000 00:00 0                          [stack]
7ffdf1578000-7ffdf157b000 r--p 00000000 00:00 0                          [vvar]
7ffdf157b000-7ffdf157d000 r-xp 00000000 00:00 0                          [vdso]
Aborted
```

</details>

In same time if launch *AzPainter* without custom locale (`$ azpainter`) there are NO such issue.

### Expected behavior

There should be no crush during initializing any floating windows.

**Comment by Symbian9** [comment]
@azelpg, any ideas? Does it fixed in your repo already?

**Comment by azelpg** [comment]
At least it didn't crash in my openbox environment, so it may have been a WM issue that I fixed in 2.16.
